code/modules/power/apc.dm

```diff
@@ -419,11 +419,14 @@
 			update_overlay |= APC_UPOVERLAY_ENVIRON2
 
 
-	. = NONE
-	if(last_update_state == update_state)
-		. |= COMSIG_ATOM_NO_UPDATE_ICON_STATE
-	if((last_update_overlay == update_overlay) && (update_state & UPSTATE_ALLGOOD))
-		. |= COMSIG_ATOM_NO_UPDATE_OVERLAYS
+	var/results = 0
+	if(last_update_state == update_state && last_update_overlay == update_overlay)
+		return 0
+	if(last_update_state != update_state)
+		results += 1
+	if(last_update_overlay != update_overlay)
+		results += 2
+	return results
```

Comment:
I cannot verbally describe how much it hurts to readd this

Reply:
APCs really need a refactor at some point.